2021q2/opath.adoc

```diff
@@ -0,0 +1,65 @@
+== VFS path descriptors API
+
+Contact: Konstantin Belousov <kib@FreeBSD.org>
+
+Unix VFS API, from very high point of view, provides two kinds of
+interfaces.  One is the path-based operations, typically manipulating
+metadata, like remove, rename, change permissions and similar.  Second
+is file-descriptor operations, typical are read, write, truncate, and
+other.  The well-known open(2) syscall returns file descriptor from
+path.
+
+The file descriptor is a reliable handle for the file which is
+targeted by operations.  Whatever other action occured to the file
+meantime, rename, even delete, file descriptor still references the
+same file, and any operation done on it, affects that file.
+
+Until recently, there were no similar reliable way to take a handle on
+the file path, and then reliably use it for the operations of the
+first kind.  Now, Linux introduced some facilities that makes it
+possible, and FreeBSD tried to provide the compatible API.
+
+First, there is a way to get the file descriptor for a path.  It is
+created with the same open(2) syscall, when the O_PATH flag is
+specified.  The returned descriptor cannot be used for normal file
+operations, it only designates a place in the filesystem namespace.
+Attempt to e.g. read or write using it result in EBADF.  But also
+system does not check the access rights on the target file, it is
+enough for the process to read the content of the directory with the
+file name.
+
+To actually use such descriptor, there is a new flag AT_EMPTY_PATH,
+understood by the *at(2) set of syscalls.  When supplied, and the path
```

Comment:
I think you may need to escape the * with \, as it looks like it's taken as markup, judging by the following text being bolded. I would check what it looks like once rendered by AsciiDoctor first, though.